File: pinyougou-parent/pingyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.pojogroup.Specification;

/****
 * 规格参数服务接口声明
 * 
 * @author warrior
 *
 */
public interface SpecificationService {

	/****
	 * 分页查询规格参数列表
	 * 
	 * @param pageNum  当前页
	 * @param pageSize 每页显示条数
	 * @return
	 */
	PageResult<TbSpecification> querySpecifications(int pageNum, int pageSize);

	/****
	 * 根据Id获取产品规格信息
	 * 
	 * @param id 产品规格Id
	 * @return
	 */
	Specification querySpecificationById(long id);

	/****
	 * 新增产品规格
	 * 
	 * @param specification 产品规格信息
	 * @return
	 */
	boolean createSpecification(Specification specification);

	/****
	 * 修改产品规格
	 * 
	 * @param specification 产品规格信息
	 * @return
	 */
	boolean updateSpecification(Specification specification);

	/****
	 * 删除产品规格信息
	 * 
	 * @param ids 产品规格Ids
	 * @return
	 */
	boolean deleteSpecifications(long[] ids);

	/****
	 * 条件查询产品规格信息
	 * 
	 * @param specification 产品规格查询条件
	 * @param pageNum       当前页
	 * @param pageSize      每页显示条数
	 * @return
	 */
	PageResult<TbSpecification> querySpecificationsByCondition(TbSpecification specification, int pageNum, int pageSize);
}

File: pinyougou-parent/pingyougou-manager-web/src/main/webapp/js/service/specificationService.js
//产品规格service 层

app.service('specificationService',function($http){
	
	//分页查询规格信息
	this.findPage=function(pageNum,pageSize){
		return $http.get('../rest/specification/collections?pageNum='+pageNum+'&pageSize='+pageSize);
	}
	
	//条件查询规格信息
	this.search=function(pageNum,pageSize,entity){
		return $http.post('../rest/specification/search?pageNum='+pageNum+'&pageSize='+pageSize,entity);
	}
	
	//新增规格信息
	this.save=function(entity){
		return $http.post('../rest/specification/create',entity);
	}
	
	//根据Id查询产品规格信息
	this.querySpecification=function(id){
		return $http.get('../rest/specification/'+id);
	}
	
	//修改规格信息
	this.updateSpecification=function(entity){
		return $http.post('../rest/specification/update',entity);
	}
	
	//删除规格信息
	this.deleteSpecifications=function(ids){
		return $http.get('../rest/specification/delete?ids='+ids);
	}
	
});